editoast/src/views/infra/mod.rs

```diff
@@ -809,6 +815,71 @@ pub async fn fetch_all_infra_states(
         .collect())
 }
 
+#[derive(Serialize, Deserialize, ToSchema)]
+struct MatchOperationalPointsForm {
+    waypoints: Vec<OperationalPointReference>,
+}
+
+/// Retrieve operational points from operational point references
+#[utoipa::path(
+    get, path = "",
+    tag = "infra",
+    params(InfraIdParam),
+    request_body = inline(MatchOperationalPointsForm),
+    responses(
+        (status = 200, description = "Operational points related to operational point references", body = Vec<OperationalPoint>)
+    ),
+)]
+async fn match_operational_points(
+    State(AppState { db_pool, .. }): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Path(InfraIdParam { infra_id }): Path<InfraIdParam>,
+    Json(operational_points_form): Json<MatchOperationalPointsForm>,
+) -> Result<Json<Vec<OperationalPoint>>> {
+    let authorized = auth
+        .check_roles([Role::OperationalStudies].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+    let mut operational_points: Vec<OperationalPoint> = vec![];
+    let conn = &mut db_pool.get().await?;
+    for waypoint in operational_points_form.waypoints {
+        operational_points.append(&mut match waypoint.reference {
+            OperationalPointIdentifier::OperationalPointId {
+                operational_point: op_id,
+            } => {
+                let ids = [op_id.0];
+                retrieve_op_from_ids(conn, infra_id, &ids)
+                    .await?
+                    .iter()
+                    .map(|tup| tup.1.schema.clone())
+                    .collect::<Vec<_>>()
+            }
+            OperationalPointIdentifier::OperationalPointDescription { trigram, .. } => {
+                let trigrams = [trigram.0];
+                retrieve_op_from_trigrams(conn, infra_id, &trigrams)
+                    .await?
+                    .iter()
+                    .flat_map(|tup| tup.1)
+                    .map(|tup| tup.schema.clone())
```

Comment:
```suggestion
                    .flat_map(|(_, op_models)| op_models)
                    .map(|(_, op_model)|op_model.schema.clone())
```

I think it's more readable to do this way.